pom.xml

```
@@ -0,0 +1,239 @@
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.apache</groupId>
+        <artifactId>apache</artifactId>
+        <version>29</version>
+    </parent>
+    <version>1.0.0-SNAPSHOT</version>
+
+    <inceptionYear>2024</inceptionYear>
+
+  <artifactId>pulsar-java-contrib</artifactId>
+  <packaging>pom</packaging>
+  <name>Pulsar Java Contrib</name>
+
+  <modules>
+      <module>pulsar-client-common-contrib</module>
+      <module>pulsar-loadbalance-contrib</module>
+      <module>pulsar-interceptor-contrib</module>
+      <module>pulsar-connector-contrib</module>
+      <module>pulsar-function-contrib</module>
+      <module>pulsar-bookkeeper-contrib</module>
+      <module>pulsar-transaction-contrib</module>
+      <module>pulsar-metrics-contrib</module>
+      <module>pulsar-auth-contrib</module>
+  </modules>
+
+  <profiles>
+    <profile>
+      <id>default</id>
+      <activation>
+        <property>
+          <name>!productized</name>
+        </property>
+      </activation>
+    </profile>
+
+    <profile>
+      <id>fullProfile</id>
+      <activation>
+        <property>
+          <name>full</name>
+        </property>
+      </activation>
+    </profile>
+  </profiles>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.projectlombok</groupId>
+            <artifactId>lombok</artifactId>
+            <version>${lombok.version}</version>
+        </dependency>
+
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-simple</artifactId>
+            <type>jar</type>
+        </dependency>
+        <dependency>
+            <groupId>org.testng</groupId>
+            <artifactId>testng</artifactId>
+            <version>${org.testing.version}</version>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>org.slf4j</groupId>
+                <artifactId>slf4j-bom</artifactId>
+                <version>${slf4j.version}</version>
+                <type>pom</type>
+                <scope>import</scope>
+            </dependency>
+            <dependency>
+                <groupId>org.apache.pulsar</groupId>
+                <artifactId>pulsar-bom</artifactId>
+                <version>${pulsar.version}</version>
+                <type>pom</type>
+                <scope>import</scope>
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
+
+    <properties>
+        <lombok.version>1.18.32</lombok.version>
+        <slf4j.version>2.0.13</slf4j.version>
+        <maven.compiler.release.version>17</maven.compiler.release.version>
```

```suggestion
        <maven.compiler.release>17</maven.compiler.release>
```
Please check the docs https://maven.apache.org/plugins/maven-compiler-plugin/examples/set-compiler-release.html